inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/service/feature/SemanticSignatureFeatureGenerator.java

```diff
@@ -0,0 +1,233 @@
+     */
+    private SemanticSignature getSemanticSignature(KnowledgeBase aKB, String aIri)
+    {
+        return semanticSignatureCache.get(new SemanticSignatureCacheKey(aKB, aIri));
+    }
+
+    private SemanticSignature loadSemanticSignature(SemanticSignatureCacheKey aKey)
+    {
+        Set<String> relatedRelations = new HashSet<>();
+        Set<String> relatedEntities = new HashSet<>();
+        try (RepositoryConnection conn = kbService.getConnection(aKey.getKnowledgeBase())) {
+            TupleQuery query = SPARQLQueryStore.generateSemanticSignatureQuery(conn,
+                    aKey.getQuery(), properties.getSignatureQueryLimit(), aKey.getKnowledgeBase());
+            try (TupleQueryResult result = query.evaluate()) {
+                while (result.hasNext()) {
+                    BindingSet sol = result.next();
+                    String propertyString = sol.getValue("p").stringValue();
+                    String labelString = sol.getValue("label").stringValue();
+                    if (propertyWithLabels != null) {
+                        Property property = propertyWithLabels.get(labelString);
+                        int frequencyThreshold = 0;
+                        boolean isBlacklisted =
+                            (propertyBlacklist != null && propertyBlacklist.contains(propertyString)
+                            || (property != null && (typeBlacklist != null
+                                && typeBlacklist.contains(property.getType()))));
+                        boolean isUnfrequent = property != null
+                            && property.getFreq() < frequencyThreshold;
+                        if (isBlacklisted || isUnfrequent) {
+                            continue;
+                        }
+                    }
+                    relatedEntities.add(labelString);
+                    relatedRelations.add(propertyString);
+                }
+            }
+            catch (Exception e) {
+                if (StringUtils.contains(e.getMessage(), "UTF-8 sequence")
+                        && !log.isDebugEnabled()) {
+                    // This is a comparatively common message - no need to always log the entire
+                    // stack trace during production, but might still be reasonable to log a
+                    // warning.
+                    log.warn("Could not get semantic signature: {}", e.getMessage());
+                }
+                else {
+                    log.error("Could not get semantic signature", e);
+                }
+            }
+        }
+
+        return new SemanticSignature(relatedEntities, relatedRelations);
+    }
+
+    private class SemanticSignatureCacheKey
+    {
+        private final KnowledgeBase knowledgeBase;
+        private final String query;
+
+        public SemanticSignatureCacheKey(KnowledgeBase aKnowledgeBase, String aQuery)
+        {
+            super();
```

Comment:
Why super?

Reply:
Auto-generated code I suppose. Some people think that calling super explicitly is a good idea.